service/src/main/kotlin/xyz/lebkuchenfm/domain/xsounds/XSoundsService.kt

```diff
@@ -45,4 +46,8 @@ class XSoundsService(private val repository: XSoundsRepository, private val file
             logger.error { "Could not increment play count for sound $soundName" }
         }
     }
+
+    suspend fun addTagToXSound(soundName: String, tag: String): Result<XSound, AddXTagToXSoundError> {
```

Comment:
We shouldn't reuse the error type from repository in the service. Those should be separate types - repository error should be for database/storage related issues and service one should concern the business logic of the action.
In this case they will be the same, but this may not be true always.